pkg/gadgets/top/ebpf/tracer/tracer.go

```diff
@@ -205,9 +205,27 @@ func (t *Tracer) nextStats() ([]types.Stats, error) {
 		if err != nil {
 			continue
 		}
-		mapSizes[curMapID] = (uint64(mapData.KeySize()) + uint64(mapData.ValueSize())) * uint64(mapData.MaxEntries())
+		defer mapData.Close()
 
-		mapData.Close()
+		f, err := os.Open(fmt.Sprintf("/proc/self/fdinfo/%d", mapData.FD()))
+		if err != nil {
+			return nil, fmt.Errorf("could not read fdinfo of map ID (%d): %w", curMapID, err)
+		}
+		defer f.Close()
+
+		sc := bufio.NewScanner(f)
+		for sc.Scan() {
+			if strings.HasPrefix(sc.Text(), "memlock:\t") {
+				lineSplit := strings.Split(sc.Text(), "\t")
+				if len(lineSplit) == 2 {
+					mapSizes[curMapID], err = strconv.ParseUint(lineSplit[1], 10, 64)
+					if err != nil {
+						return nil, fmt.Errorf("could not read memlock of map ID (%d): %w", curMapID, err)
+					}
+					break
+				}
+			}
+		}
 	}
```

Comment:
How about putting all of this in a helper function? `func getMemoryUsage(m *ebpf.Map)`?

Reply:
Done.
I should do that way more often